src/python/pants/source/source_root.py

```diff
@@ -301,33 +254,60 @@ def register_options(cls, register):
             fingerprint=True,
             default=cls._DEFAULT_THIRDPARTY_ROOTS,
             advanced=True,
+            removal_version="1.30.0.dev0",
+            removal_hint="Use --roots instead.",
             help=fixed_help_fmt.format("third-party"),
         )
 
+        register(
+            "--roots",
+            metavar="<list>",
+            type=list,
+            fingerprint=True,
+            # For Python a good default might be the repo root, but that would be a bad default
+            # for other languages, so best to have no default for now, and force users to be
+            # explicit about this when integrating Pants.  It's a fairly trivial thing to do.
+            default=[],
+            advanced=True,
+            help="A list of source root suffixes. A directory with this suffix will be considered "
+            "a potential source root. E.g., `src/python` will match `<buildroot>/src/python`, "
+            "`<buildroot>/project1/src/python` etc. Prepend a `^/` to anchor the match at the "
+            "buildroot.  E.g., `^/src/python` will match `<buildroot>/src/python` but not "
+            "`<buildroot>/project1/src/python`.",
```

Comment:
How do you have a top level build root? Simply `""`, right? Would be good to document.

Reply:
Maybe also link to https://pants.readme.io/docs/source-roots. That URL should be stable.